### DISCUSSION TITLE
WalletConnectModal filter wallets by supported target chain

### DISCUSSION THREAD

**Discussion by gonzamontiel** [WalletConnect/walletconnect-monorepo#3282] [Developer Support] [answered]
We are using WalletConnectModal (v 2.x) along with our React application. What we are trying to achieve is to, instead of fetching all wallets from the WalletConnect explorer, filter those that support a specific chain (like, in our case, Substrate based chains). This option seemed to be called `chains` , but I cannot longer see it available in WalletConnectModalConfig.
I was wondering if this is even possible, I've been also playing around with `wagmi` documentation but didn't get a clear answer.

**Comment by glitch-txs** [comment]
Hi, is this option helpful? You can filter the wallets that are shown in the modal
https://docs.walletconnect.com/2.0/web/web3modal/react/wagmi/options#explorerrecommendedwalletids-optional

**Reply by gonzamontiel** [reply]
Hey @glitch-txs , thanks for your answer. So I thought about combining `explorerExcludedWalletIds` (set to ALL) and `explorerRecommendedWalletIds` (set to the desired ones).
But of course this means to maintain our own list of supported Substrate wallets, while the question aimed more to automatically filter wallets by supported chain. Do you know if this is possible?

**Reply by glitch-txs** [reply, marked as answer]
Hi @gonzamontiel we don't have this feature in v2 at the moment, I think however it would be a great feature and we'll consider adding it to Web3Modal v3. Thanks for the feedback 👍

**Reply by gonzamontiel** [reply]
hey, thanks for your answer, looking forward to seeing the next version  :)
